lib/mobile_application_platform/security_token/service.rb

```diff
@@ -16,9 +16,23 @@ def token(application:, icn:)
         Rails.logger.info("#{config.logging_prefix} token success, application: #{application}, icn: #{icn}")
         parse_response(response, application, icn)
       rescue Common::Client::Errors::ClientError => e
+        description = case e.status
+                      when 500
+                        'service returned Internal Server error most likely due to an invalid ICN'
+                      when 401, 400
+                        if e.body.is_a?(Hash) && e.body&.key?(:error)
+                          e.body[:error]
+                        elsif e.body.is_a?(Hash) && e.body&.key?(:error_description)
+                          e.body[:error_description]
+                        else
+                          'unknown'
+                        end
+                      else
+                        'unknown'
+                      end
```

Comment:
I think this is a great opportunity for pattern matching. Here's what this would look like, using pattern matching to clean it up:

```ruby
description = case e
              in _msg, 500, _body
                'service returned Internal Server error most likely due to an invalid ICN'
              in _msg, (400|401), { error: }
                error
              in _msg, (400|401), { error_description: }
                error_description
              else
                'unknown'
              end
```

In order for this to work, the error must respond to `deconstruct`. This is pretty easy to do, won't mess anything up, and will allow other devs to pattern match off instances of this error class in the future. You can just implement the `deconstruct` method in the `Common::Client::Errors::ClientError` class like so:

```ruby
module Common
  module Client
    module Errors
      class ClientError < Error
      ...
      def deconstruct
        [message, status, body]
      end
    end
  end
end
```